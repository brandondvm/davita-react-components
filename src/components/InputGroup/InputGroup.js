import React from 'react';
import PropTypes from 'prop-types';
import idx from 'idx';

import Input from '../Input/Input';

const InputGroup = ({ items,  handlers, register, errors, watch }) => {

  const getInput = item => {
    const { id, name, type, label, selectButtons, placeholder, validation, accessibiltity, innerLabel } = item;
    const error = idx(errors, _ => _[name]);
    const selectButtonsProps = selectButtons ? {
      leftButtonLabel: selectButtons.leftButtonLabel,
      leftButtonOnClick: handlers[selectButtons.leftButtonOnClick],
      rightButtonLabel: selectButtons.rightButtonLabel,
      rightButtonOnClick: handlers[selectButtons.rightButtonOnClick]
    } : null;

    return (
      <Input
        register={register(validation)}
        selectButtons={selectButtonsProps}
        errorMessage={error?.type ? error.message : null}
        placeholder={placeholder}
        type={type}
        label={label}
        id={id}
        innerLabel={innerLabel}
        name={name}
        accessibiltity={accessibiltity}
        defaultValue={watch(name)}
      />
    )
  };

  return (
    <div className="input-group-component">
      { 
        items.map((item) => (
          <React.Fragment key={item.id}>
            { getInput(item) }
            <span className="input-group-component_divider">/</span>
          </React.Fragment>
        ))
      }
    </div>
  );
};

InputGroup.propTypes = {
  items: PropTypes.array,
  handlers: PropTypes.object,
  register: PropTypes.func,
  errors: PropTypes.object,
};

export default InputGroup;
