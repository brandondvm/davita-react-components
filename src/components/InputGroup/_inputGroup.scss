.input-group-component {
  display: flex;

  &_divider {
    font-size: 16px;
    margin: 13px 15px 0;
    &:last-of-type {
      display: none;
    }
  }

  .input-component {
    width: 100%;
  }
}
