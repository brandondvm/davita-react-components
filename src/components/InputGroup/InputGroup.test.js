import enzyme from 'enzyme';
import React from 'react';

import InputGroup from './InputGroup';

const mockProps = {
  items: [
    {
      id: 'systolic',
      name: 'systolic',
      label: 'Systolic',
      placeholder: 'Systolic',
      componentType: 'input',
      innerLabel: 'mmHg',
      type: 'number',
      validation: {
        required: 'Systolic value is required',
      }
    },
    {
      id: 'diastolic',
      name: 'diastolic',
      label: 'Diastolic',
      placeholder: 'Diastolic',
      componentType: 'input',
      innerLabel: 'mmHg',
      type: 'number',
      validation: {
        required: 'Diastolic value is required',
      }
    },        
  ], 
  handlers: {},
  register: jest.fn(),
  watch: jest.fn(),
  errors: {}
};

function setup(passedProps = {}) {
  const updatedProps = {
    ...mockProps,
    ...passedProps,
  };

  return enzyme.shallow(<InputGroup { ...updatedProps } />);
}

describe('InputGroup Component', () => {
  test('Component renders correctly', () => {
    const wrapper = setup();
    expect(wrapper.get(0)).toMatchSnapshot();
  });

  test('Component should render inputs correctly on DOM', () => {
    const wrapper = setup();
    const inputComponents = wrapper.find('Input');
    expect(inputComponents.length).toEqual(2);
  });
});
