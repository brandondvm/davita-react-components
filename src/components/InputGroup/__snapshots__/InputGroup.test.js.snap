// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputGroup Component Component renders correctly 1`] = `
<div
  className="input-group-component"
>
  <React.Fragment>
    <Input
      errorMessage={null}
      id="systolic"
      innerLabel="mmHg"
      label="Systolic"
      name="systolic"
      placeholder="Systolic"
      selectButtons={null}
      type="number"
    />
    <span
      className="input-group-component_divider"
    >
      /
    </span>
  </React.Fragment>
  <React.Fragment>
    <Input
      errorMessage={null}
      id="diastolic"
      innerLabel="mmHg"
      label="Diastolic"
      name="diastolic"
      placeholder="Diastolic"
      selectButtons={null}
      type="number"
    />
    <span
      className="input-group-component_divider"
    >
      /
    </span>
  </React.Fragment>
</div>
`;
